nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,613 @@
+                            invalidFF.set(session.append(invalidFF.get(), new OutputStreamCallback() {
+                                @Override
+                                public void process(OutputStream out) throws IOException {
+                                    out.write(print(headerList, csvPref, isFirstLine.get()));
+                                }
+                            }));
+                            validFF.set(session.append(validFF.get(), new OutputStreamCallback() {
+                                @Override
+                                public void process(OutputStream out) throws IOException {
+                                    out.write(print(headerList, csvPref, isFirstLine.get()));
+                                }
+                            }));
+                            isFirstLine.set(false);
+                        }
+                    }
+
+                    boolean stop = false;
+
+                    while (!stop) {
+                        try {
+
+                            final List<Object> list = listReader.read(cellProcs);
+                            stop = list == null;
+
+                            if(!isWholeFFValidation && !stop) {
+                                validFF.set(session.append(validFF.get(), new OutputStreamCallback() {
+                                    @Override
+                                    public void process(OutputStream out) throws IOException {
+                                        out.write(print(list, csvPref, isFirstLine.get()));
+                                    }
+                                }));
+                                okCount.set(okCount.get() + 1);
+
+                                if(isFirstLine.get()) {
+                                    isFirstLine.set(false);
+                                }
+                            }
+
+                        } catch (final SuperCsvCellProcessorException e) {
+                            valid.set(false);
+                            if(isWholeFFValidation) {
+                                logger.debug("Failed to validate {} against schema due to {}; routing to 'invalid'", new Object[]{flowFile}, e);
+                                break;
+                            } else {
+                                // we append the invalid line to the flow file that will be routed to invalid relationship
+                                invalidFF.set(session.append(invalidFF.get(), new OutputStreamCallback() {
+                                    @Override
+                                    public void process(OutputStream out) throws IOException {
+                                        out.write(print(e.getCsvContext().getRowSource(), csvPref, isFirstLine.get()));
+                                    }
+                                }));
+
+                                if(isFirstLine.get()) {
+                                    isFirstLine.set(false);
```

Comment:
Currently valid and invalid share the same "isFirstLine". So if the first row is valid and the second is invalid the invalid FF will be prepended with the new line character(s)